### PR TITLE
refactor sync function and improve general logging

### PR DESCRIPTION
**Pull request by shenmo3**
## Description
Several minor changes aimed to improve code and log message readability.

## Changes
1. Changed addrGroup notify log to stay consistent with appliedToGroup.
2. On syncWithCloud, don't recompute np rules if np rule is ready.
3. Improve VMP message
4. Refactor the appliedTo sync function, split logic into separate functions.

### CODE REVIEW

File: pkg/controllers/cloud/networkpolicy_sync.go [hunk 1]
```diff
@@ -365,3 +279,82 @@ func (r *NetworkPolicyReconciler) getNICsOfCloudResources(resources []*securityg
...
+	for _, ip := range eRule.ToDstIP {
+		updateCountForItem(ip.String(), items, subtract)
+	}
+	for _, sg := range eRule.ToSecurityGroups {
+		updateCountForItem(sg.String(), items, subtract)
+	}
+}
+
+// updateCountForItem adds or subtracts the item count in the items map.
+func updateCountForItem(item string, items map[string]int, subtract bool) {
```
Comment (reachjainrahul) [inline pkg/controllers/cloud/networkpolicy_sync.go:354]: do it inplace.. no need for new function

**Comment by reachjainrahul** [comment]
/nephe-test-e2e-kind, /nephe-test-e2e-eks, /nephe-test-e2e-aks

File: pkg/apiserver/registry/virtualmachinepolicy/rest.go [hunk 2]
```diff
@@ -134,7 +134,7 @@ func (r *REST) convertToVMP(internal *cloud.NetworkPolicyStatus) *runtimev1alpha
 	inProgress := false
 	npStatusList := make(map[string]*runtimev1alpha1.NetworkPolicyStatus)
 	for anp, status := range internal.NPStatus {
-		if status == cloud.NetworkPolicyStatusApplied {
+		if strings.Contains(status, cloud.NetworkPolicyStatusApplied) {
```
Comment (reachjainrahul) [inline pkg/apiserver/registry/virtualmachinepolicy/rest.go:137]: Is the get vmp output updated with this change?

Comment (shenmo3) [reply pkg/apiserver/registry/virtualmachinepolicy/rest.go:137]: no, still shown as SUCCESS

**Review by reachjainrahul — APPROVED** [review]
/LGTM

**Pull request merged**